### ISSUE TITLE
Truffle develop environment error: Error encountered, bailing. Network state unknown. Review successful transactions manually. Error: VM Exception while processing transaction: out of gas

### ISSUE THREAD

**Issue by DOkwufulueze** [trufflesuite/truffle#669]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

I am having the `Error: VM Exception while processing transaction: out of gas` error in truffle develop environment when trying to deploy my contracts. I have tried running truffle develop with `-l 0xfffffffffff` [I also tried with base 10 numbers] in order to increase the gas amount, but the error continues. Based on a suggestion on gitter, I also added solc configuration to enable solc optimser in `truffle.js`, but the error remains the same. Whenever the migration fails, gas usage is always at 6721975. Is there a way to increase gas amount in truffle develop?

## Steps to Reproduce

Enable solc optimiser in truffle.js:
```js
solc: {
  optimizer: {
    enabled: true,
    runs: 200
  }
}
```
`truffle develop`
`migrate`

## Expected Behavior

Migrations are supposed to run without any error.

## Actual Results

On truffle develop environment
```js
Error encountered, bailing. Network state unknown. Review successful transactions manually.
Error: VM Exception while processing transaction: out of gas
```

On truffle develop --log
```js
develop:testrpc   Transaction: 0xded244790b00b1dac381d1357998ef97685571df864005b5f164ae34a7e7073e +0ms
develop:testrpc   Contract created: 0x0493e283832fd4265e366256be6652ebd06fc0bd +0ms
develop:testrpc   Gas usage: 6721975 +0ms
develop:testrpc   Block Number: 8 +0ms
develop:testrpc   Block Time: Mon Nov 06 2017 14:11:43 GMT+0100 (WAT) +1ms
develop:testrpc   Runtime Error: out of gas +0ms
develop:testrpc  +0ms
```
## Environment

* Operating System: Ubuntu 16.04 LTS
* Truffle version: 4.0.0
* Ethereum client: Geth 1.6.6-stable
* node version: v8.5.0
* npm version: 5.3.0

**Comment by gnidan** [comment]
@DOkwufulueze did you delete your `build/contracts/` directory after turning the optimizer back on? That might be the problem, but please let me know if that doesn't solve it. Thanks!

**Comment by DOkwufulueze** [comment]
@gnidan, thanks for the response. Yes, I deleted the `build/contracts/` directory.

I have stated on gitter, my realisation that the only way to have a successful deployment is to split the contract into separate contracts. And @tcoulter has responded with news of the update on the truffle release page discussing the security concerns about turning solc optimiser on, and advising the split option if possible.

So for now, I'm simply using the split option, and all is well again.

Thanks again.

I'll be closing this issue since it's resolved.
